modules/ROOT/pages/functions.adoc

```diff
@@ -0,0 +1,143 @@
+
+Python functions are added by loading a Python file (.py) or a zipped Python file (.zip). When adding Python files, the Class Name is specified as the name of the Python file without the extension plus the class you want to execute. 
+
+For example, if the Python file is called `function.py` and the class is `exclamation`, then the class name would be `function.exclamation`. The file can contain multiple classes, but only one will be used. If there is no class in the Python file (when using a basic function, for example) just specify the filename without the extension (ex. `function`).
+
+Java functions are added by loading a Java jar file (.jar). When adding Java files, you also need to specify the name of the class to execute as the function. 
+[start=4]
+. Your input topics, output topics, log topics, and processing guarantees will auto-populate. 
+
+. Provide a *Configuration Key* in the dropdown menu. See the https://pulsar.apache.org/functions-rest-api/#[Pulsar Docs] for a list of configuration keys. 
+
+. Select *Add* to add your function. 
+
+You have created a function for this namespace. You can confirm your function was created in the *Manage* tab.
+
+== Managing your function
+
+You can start, stop, and restart your function by selecting *Manage* in the *Functions* dashboard. 
+
+image::admin-console-function-added.png[Manage Functions]
+
+=== Monitoring your function
+
+Functions produce logs to help you in debugging. To view your function's logs, select *Logs* in the *Manage* dashboard. 
+
+image::admin-console-logs.png[Function Log]
+
+=== Updating your function
+
+A function that is already running can be updated with new configuration. The following settings can be updated:
+
+* Function code
+* Output topic
+* Log topic
+* Number of instances
+* Configuration keys
+
+If you need to update any other setting of the function, delete and then re-add the function.
+
+To update your function, select *Update* in the *Manage* dashboard. 
+
+=== Deleting your function
+
+To delete a function, select *Delete* in the *Manage* dashboard. 
+
+A *Function-name Deleted Successfully!* flag will appear to let you know you've deleted your function.
+
+=== Triggering your function
+
+Triggering a function is a convenient way to test that the function is working. When you trigger a function, you are publishing a message on the function’s input topic, which triggers the function to run. 
+
+To trigger a function, select *Trigger* in the *Manage* dashboard.
+
+image::admin-console-trigger-function.png[Trigger Function]
+
+Enter your message in the *Message to Send* field, and select the output topic. In this case, the trigger sends the string `Hello world!` to your exclamation function with no output function. If the function has an output topic and the function returns data to the output topic, it will be displayed. 
+
+== Add function with Pulsar CLI 
```

Comment:
I would recommend to move this section before the section of "Add Functions n Luna Streaming"